builtin/providers/aws/resource_aws_codecommit_trigger.go

```diff
@@ -0,0 +1,170 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/codecommit"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsCodeCommitTrigger() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsCodeCommitTriggerCreate,
+		Read:   resourceAwsCodeCommitTriggerRead,
+		Delete: resourceAwsCodeCommitTriggerDelete,
+
+		Schema: map[string]*schema.Schema{
+			"repository_name": &schema.Schema{
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+					value := v.(string)
+					if len(value) > 100 {
+						errors = append(errors, fmt.Errorf(
+							"%q cannot be longer than 100 characters", k))
+					}
+					return
+				},
+			},
+
+			"trigger": &schema.Schema{
+				Type:     schema.TypeSet,
+				ForceNew: true,
+				Required: true,
+				MaxItems: 10,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+
+						"destination_arn": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+
+						"custom_data": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+
+						"branches": &schema.Schema{
+							Type:     schema.TypeList,
+							Optional: true,
+							ForceNew: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+
+						"events": &schema.Schema{
+							Type:     schema.TypeList,
+							Required: true,
+							ForceNew: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceAwsCodeCommitTriggerCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).codecommitconn
+	region := meta.(*AWSClient).region
+
+	//	This is a temporary thing - we need to ensure that CodeCommit is only being run against us-east-1
+	//	As this is the only place that AWS currently supports it
+	if region != "us-east-1" {
```

Comment:
Let's remove this now - we should add a note to the docs pointing to where it is used. This will mean that when it is rolled to another region, users can't use it. Please also remove from `aws_code_commit` as well :)